test/minishift/test_suite.go

```diff
@@ -0,0 +1,172 @@
+	"github.com/fabric8-services/fabric8-tenant/test"
+	"github.com/fabric8-services/fabric8-tenant/test/gormsupport"
+	"github.com/fabric8-services/fabric8-tenant/test/resource"
+	"github.com/fabric8-services/fabric8-tenant/test/stub"
+	"github.com/satori/go.uuid"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"net/http"
+	"sync"
+	"testing"
+	"time"
+)
+
+// TestSuite is a base for tests using Minishift and gorm db
+type TestSuite struct {
+	gormsupport.DBTestSuite
+	ClusterService  *stub.ClusterService
+	AuthService     *stub.AuthService
+	Config          *configuration.Data
+	toReset         func()
+	minishiftConfig *Data
+}
+
+func (s *TestSuite) SetupTest() {
+	resource.Require(s.T(), resource.Database)
+
+	config, err := NewData()
+	require.NoError(s.T(), err)
+	if config.GetMinishiftURL() == "" {
+		s.T().Skipf("Minishift URL using a variable [%s] has to be set", varMinishiftURL)
+	}
+
+	s.minishiftConfig = config
+
+	s.DBTestSuite.SetupTest()
+	s.Config, s.toReset = prepareConfig(s.T())
+
+	log.InitializeLogger(s.Config.IsLogJSON(), s.Config.GetLogLevel())
+
+	s.ClusterService = &stub.ClusterService{
+		APIURL: s.minishiftConfig.GetMinishiftURL(),
+		User:   s.minishiftConfig.GetMinishiftAdminName(),
+		Token:  s.minishiftConfig.GetMinishiftAdminToken(),
+	}
+	s.AuthService = &stub.AuthService{
+		OpenShiftUsername:  s.minishiftConfig.GetMinishiftUserName(),
+		OpenShiftUserToken: s.minishiftConfig.GetMinishiftUserToken(),
+	}
+}
+
+func (s *TestSuite) TearDownTest() {
+	s.DBTestSuite.TearDownTest()
+	s.toReset()
+}
+
+func (s *TestSuite) GetClusterService() cluster.Service {
+	return s.ClusterService
+}
+
+func (s *TestSuite) GetAuthService(tenantID uuid.UUID) auth.Service {
+	s.AuthService.TenantID = tenantID
+	return s.AuthService
+}
+
+func (s *TestSuite) GetConfig() *configuration.Data {
+	return s.Config
+}
+
+func prepareConfig(t *testing.T) (*configuration.Data, func()) {
+	resetVars := test.SetEnvironments(
+		test.Env("F8_AUTH_TOKEN_KEY", "foo"),
+		test.Env("F8_API_SERVER_USE_TLS", "false"),
+		test.Env("F8_LOG_LEVEL", "error"),
+		test.Env("F8_KEYCLOAK_URL", "http://keycloak.url.com"))
+	config, resetConf := test.LoadTestConfig(t)
+	reset := func() {
+		resetVars()
+		resetConf()
+	}
+	return config, reset
+}
+
+func VerifyObjectsPresence(t *testing.T, mappedObjects map[string]environment.Objects, options openshift.ApplyOptions, version string) {
+	size := 0
+	for _, objects := range mappedObjects {
+		size += len(objects)
+	}
+	errorChan := make(chan error, size)
+	defer func() {
+		close(errorChan)
+		for err := range errorChan {
+			assert.NoError(t, err)
+		}
+	}()
+
+	var wg sync.WaitGroup
+	for ns, objects := range mappedObjects {
+		for _, obj := range objects {
+			wg.Add(1)
+			go func(obj environment.Object, ns string) {
+				defer wg.Done()
+				errorChan <- test.WaitWithTimeout(1 * time.Minute).Until(objectIsUpToDate(obj, ns, options, version))
+			}(obj, ns)
+		}
+	}
+	wg.Wait()
```

Comment:
maybe worth to use timeout instead of waiting on wg done, otherwise we'll have a deadlock

Reply:
agreed, I don't see why there would be a deadlock here.

Reply:
Hmm I missed it. Thanks for clarification